File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    answer = []
    a = []
    if m > n:
        n, m = m, n
    for i in range(1,n+1):
        if m % i == 0 and n % i == 0:
            a.append(i)   
    answer.append(max(a))
    answer.insert(1,int(n*m/max(a)))
    return answer

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.*;

class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        ArrayList<Integer> a = new ArrayList<>();
        if(m > n){
            int k = n;
            n = m;
            m = k;
        }
        for(int i = 1; i <= n; i++){
            if(m % i == 0 && n % i == 0){
                a.add(i);
            }
        }
        int max = 0;
        for (int i : a) {
            if (i > max) {
                max = i;
            }
        }
        answer[0] = max;
        answer[1] = (n * m) / max;
        return answer;
    }
}